src/services/resolver/GithubAliasResolver.php

```diff
@@ -68,7 +68,7 @@ private function localResolve(string $name): GithubRepository {
         try {
             $file = $this->fileDownloader->download($url);
         } catch (DownloadFailedException $e) {
-            $this->updateRateLimit();
+            $this->updateRateLimit($this->rateLimit);
```

Comment:
I'm not sure I understand this change.

Why would we pass on property of this class as a parameter?

Reply:
The problem came from `updateRateLimit()`:
the code doesn't guarantee that `$this->rateLimit` is not null when the method is called, so Psalm detects a potential error.

I've passed `$this->rateLimit` as an argument because in this context `ensureWithinRateLimit()` is called before and, thanks to its `@psalm-assert` annotation, Psalm understands that rateLimit is not null.
I thought this solution was elegant but I could find another way to make Psalm happy.

For instance, we could have something like that:
```php
private function updateRateLimit(): void {
    // [...]

    assert($this->rateLimit !== null);

    $this->rateLimit = new RateLimit(
        $this->rateLimit->getLimit(),
        // [...]
```

I tend to avoid the usage of `assert()`, but it's just a personal preference.

Reply:
I personally like assert a lot because it's actual code rather than docblocks like `/** @var ... `. But I see your point. Let's keep your solution.